test/envtest/specific_gateway_envtest_test.go

```diff
@@ -29,49 +30,230 @@ func TestSpecificGatewayNN(t *testing.T) {
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	gw := deployGateway(ctx, t, ctrlClient)
-	ignoredGW := deployGateway(ctx, t, ctrlClient)
+
+	var (
+		gw, gwc = deployGateway(ctx, t, ctrlClient)
+		nn      = client.ObjectKeyFromObject(&gw)
+		// We use the same GatewayClass here.
+		gwIgnored = deployGatewayUsingGatewayClass(ctx, t, ctrlClient, gwc)
+		nnIgnored = client.ObjectKeyFromObject(&gwIgnored)
+	)
+
 	RunManager(ctx, t, envcfg,
 		AdminAPIOptFns(),
 		WithPublishService(gw.Namespace),
 		WithGatewayFeatureEnabled,
 		WithGatewayAPIControllers(),
-		WithGatewayToReconcile(fmt.Sprintf("%s/%s", gw.Namespace, gw.Name)),
+		WithGatewayToReconcile(nn.String()),
 	)
 
-	createHTTPRoutes(ctx, t, ctrlClient, gw, 1)
-	createHTTPRoutes(ctx, t, ctrlClient, ignoredGW, 1)
+	routeCount := 10
+	routes := createHTTPRoutes(ctx, t, ctrlClient, gw, routeCount)
+	ignoredRoutes := createHTTPRoutes(ctx, t, ctrlClient, gwIgnored, routeCount)
+
+	t.Run("configured specific gateway gets its listener status filled", func(t *testing.T) {
+		require.Eventually(t, func() bool {
+			err := ctrlClient.Get(ctx, nn, &gw)
+			if err != nil {
+				t.Logf("Failed to get gateway %s: %v", nn, err)
+				return false
+			}
+			httpListener, ok := lo.Find(gw.Status.Listeners, func(listener gatewayapi.ListenerStatus) bool {
+				return listener.Name == "http"
+			})
+			if !ok {
+				t.Logf("Failed to find http listener status in gateway %s", nn)
+				return false
+			}
+			if httpListener.AttachedRoutes != int32(routeCount) {
+				t.Logf("Expected %d routes to be attached to the http listener, got %d", routeCount, httpListener.AttachedRoutes)
+				return false
+			}
+
+			return true
+		}, waitTime, tickTime, "Failed to attach route to gateway")
+	})
+	t.Run("HTTPRoute attached to configured specific gateway gets its status parent filled", func(t *testing.T) {
+		require.Eventually(t, func() bool {
+			route := gatewayapi.HTTPRoute{}
+			routeNN := client.ObjectKeyFromObject(routes[0])
+			err := ctrlClient.Get(ctx, routeNN, &route)
+			if err != nil {
+				t.Logf("Failed to get route %s: %v", routeNN, err)
+				return false
+			}
+			for _, p := range route.Status.Parents {
+				if lo.ContainsBy(p.Conditions, func(c metav1.Condition) bool {
+					return c.Type == string(gatewayapi.RouteConditionAccepted) && c.Status == metav1.ConditionTrue
+				}) {
+					return true
+				}
```

Comment:
```suggestion
				return lo.ContainsBy(p.Conditions, func(c metav1.Condition) bool {
					return c.Type == string(gatewayapi.RouteConditionAccepted) && c.Status == metav1.ConditionTrue
				})
```
Will it be the same, won't it?

Reply:
Yup it does seem like it. I thought  there could be something else I'd put  after this check but I didn't. Let's change it as suggested.